gdx/src/com/badlogic/gdx/graphics/g2d/SpriteBatch.java

```diff
@@ -38,7 +38,14 @@
 public class SpriteBatch implements Batch {
 	/** @deprecated Do not use, this field is for testing only and is likely to be removed. Sets the {@link VertexDataType} to be
 	 *             used when gles 3 is not available, defaults to {@link VertexDataType#VertexArray}. */
-	@Deprecated public static VertexDataType defaultVertexDataType = VertexDataType.VertexArray;
+	@Deprecated public static VertexDataType defaultVertexDataType = VertexDataType.VertexBufferObject;
+
+	/** Used to completely override the vertex type used by SpriteBatch. This is useful for picking a specific vertex data type on
+	 * construction of the sprite batch. Recommended to reset this back to defaultVertexDataType Once the batch has been created
+	 * with this flag */
+	@Deprecated public static VertexDataType overrideVertexType = null;
```

Comment:
OK, another Deprecated field; I can sorta see why this exists now. Is there a plan to eventually remove these public deprecated fields?

Reply:
These are just for warnings mostly, as you should not touch these without understanding the consequences. There is no safe handling of what is compatible, no errors etc.

Probably would be better to have extra SpriteBatch constructors to create with specific VertexDataTypes, and then handle all incompatibilities there and give sensible warnings and exceptions when someone tries to create with something invalid.

Something that can be addressed in a separate PR as wanted this to be as noninvasive as possible.